keanu-examples/bayesianMethodsForHackers/src/main/java/com/examples/TextMessaging.java

```diff
@@ -0,0 +1,82 @@
+package com.examples;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.util.csv.ReadCsv;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.binary.compare.GreaterThanVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.ExponentialVertex;
+import io.improbable.keanu.vertices.generic.nonprobabilistic.If;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.intgr.probabilistic.PoissonVertex;
+import io.improbable.keanu.vertices.intgr.probabilistic.UniformIntVertex;
+
+/**
+ * When did the author's text messaging rate increase, based on daily messaging counts?
+ * <p>
+ * http://nbviewer.jupyter.org/github/CamDavidsonPilon/Probabilistic-Programming-and-Bayesian-Methods-for-Hackers/blob/master/Chapter1_Introduction/Ch1_Introduction_PyMC3.ipynb#Example:-Inferring-behaviour-from-text-message-data
+ */
+public class TextMessaging {
+
+    public static TextMessagingResults run() {
+
+        final TextMessagingData data = ReadCsv.fromResources("text_messaging_data.csv")
+            .asVectorizedColumnsDefinedBy(TextMessagingData.class)
+            .load();
+
+        final int numberOfDays = (int) data.numberOfMessages.getLength();
+        // These hyperparameters differ from the alpha used in the example book
+        // This is because the sampling algorithm of choice uses the prior distribution
+        // as its proposal distribution. The suggested parameters were too wide, resulting
+        // in bad proposals and by extension bad samples.
+        // When it is easier to decouple the prior from the proposal distribution, we should revisit this
+        final double alpha = 10;
+
+        ExponentialVertex earlyRate = new ExponentialVertex(alpha);
```

Comment:
Should the early rate and late rate have the same `alpha`?

Reply:
That's what the example shows...